pkg/provisioner/ironic/ironic.go

```diff
@@ -510,8 +510,13 @@ func (p *ironicProvisioner) ValidateManagementAccess(data provisioner.Management
 	driverInfo := bmcAccess.DriverInfo(p.bmcCreds)
 	// FIXME(dhellmann): We need to get our IP on the
 	// provisioning network from somewhere.
-	driverInfo["deploy_kernel"] = deployKernelURL
-	driverInfo["deploy_ramdisk"] = deployRamdiskURL
+	if deployKernelURL != "" && deployRamdiskURL != "" {
+		driverInfo["deploy_kernel"] = deployKernelURL
+		driverInfo["deploy_ramdisk"] = deployRamdiskURL
+	}
+	if deployISOURL != "" {
+		driverInfo["deploy_iso"] = deployISOURL
+	}
```

Comment:
It would be nice to cover this part with a small test

Reply:
I think this code is not unit tested at all, but I'll double-check.

Reply:
Yeah it could be, we started to cover the `ValidateManagementAccess` some time ago but there were still some gap. A test for this part could also serve as a useful documentation for the allowed cases.

Reply:
I also checked and looks like there's not a direct way to check, probably the mock will need some additional retouch (in case `TestValidateManagementAccessCreateWithLiveIso` could be a good starting point to look at).